src/main/kotlin/com/terraformation/backend/support/atlassian/resource/AtlassianResource.kt

```diff
@@ -0,0 +1,17 @@
+package com.terraformation.backend.support.atlassian.resource
+
+import com.terraformation.backend.support.atlassian.AtlassianHttpClient
+import io.ktor.client.request.HttpRequestBuilder
+import io.ktor.client.statement.HttpResponse
+
+/**
+ * Every Atlassian API call will implement this interface. The intended workflow is that every time
+ * an API call is made, an instance of this interface will be created. The [AtlassianHttpClient]
+ * will invoke the [buildRequest] method so the caller will not be able to erroneously or
+ * maliciously set the request body.
+ */
+interface AtlassianResource<T> {
```

Comment:
From the names, it wasn't clear to me what the difference between a "resource" and a "model" was. Would it make sense to rename this to `AtlassianRequest` and rename its package to `request` given that the word "request" is used in a couple of the implementation classes already? And then maybe make sure all the implementation names end with "Request" for consistency.

Reply:
Was hoping to follow Jira documentation phrases. I do think requests or HTTP request is a better name. Will rename for consistency.